### PR TITLE
DIDN exact recon - New PR

### PR DESCRIPTION
**Pull request by kevinmicha**
Now done from `dev` branch instead of `master`.

### CODE REVIEW

File: gray_model.py [hunk 1]
```diff
@@ -208,5 +209,8 @@ def forward(self, x):
         out= self.subpixel(out)
         out = self.conv_output(out)
         out = torch.add(out, residual)
```
Comment (zaccharieramzi) [inline gray_model.py]: Here watch out for the residual

File: gray_model.py [hunk 2]
```diff
@@ -208,5 +209,8 @@ def forward(self, x):
...
+        noise_std = torch.reshape(noise_std, (residual.size()[0], 1, 1, 1))
+        if self.exact_recon:
+            out = torch.subtract(residual, noise_std * out)
```
Comment (zaccharieramzi) [inline gray_model.py]:
```suggestion
            out = residual - noise_std * out
```

**Review by zaccharieramzi — CHANGES_REQUESTED** [review]
Same as on the other PR

File: gray_model.py (hunk 2 above)
Comment (kevinmicha) [reply gray_model.py]: done

File: gray_model.py (hunk 1 above)
Comment (kevinmicha) [reply gray_model.py]: fixed

**Review by zaccharieramzi — APPROVED** [review]
LGTM

**Comment by kevinmicha** [comment]
Merging

**Pull request merged**